core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderService.js

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for rendering learner action HTML strings.
+ */
+
+oppia.factory('LearnerActionRenderService', [
+  'ACTION_TYPE_ANSWER_SUBMIT', 'ACTION_TYPE_EXPLORATION_START',
+  'ACTION_TYPE_EXPLORATION_QUIT', 'ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS',
+  function(
+      ACTION_TYPE_ANSWER_SUBMIT, ACTION_TYPE_EXPLORATION_START,
+      ACTION_TYPE_EXPLORATION_QUIT, ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS) {
+    var renderExplorationStartActionHTML = function(stateName) {
+      var htmlString =
+        'Started exploration at card "' + stateName + '".<br><br>';
```

Comment:
Hmm, not sure I like the idea of keeping these `<br>` tags here. This looks like a job better suited for CSS:

```javascript
'<span class="learner-action">Started exploration at card "' + stateName + '".</span>'
```

This way we simply add a margin to the `learner-action` class to add spacing.

Reply:
Done.